### DISCUSSION TITLE
How to Prevent Cameras from Capturing Objects in Other Environments?

### DISCUSSION THREAD

**Discussion by johnnylu305** [isaac-sim/IsaacLab#552] [Q&A]
Hi,

I followed the CartPole example to create a large environment composed of multiple smaller environments for RL parallelism training. In each small environment, I have a drone flying and capturing an object with a camera. My problem is that the drone may capture objects from other small environments. Is there any way to avoid this? Additionally, how can I make each small environment independent so that they don't affect each other in terms of perception or collisions?

Thank you!

**Comment by ahanjaya** [comment]
I also need this

**Comment by hutslib** [comment]
I also need this

**Comment by WangYCheng23** [comment]
i also need that for manipulation scene

**Comment by Mayankm96** [comment]
There isn't a mechanism for it in Isaac Sim. I am not 100% sure when this feature will be available.

As a workaround, you can add walls to your environment so the camera doesn't see prims beyond that environment instance.

**Reply by johnnylu305** [reply]
Thanks for your response. I've previously tried adding cubes as walls in the environment. Everything worked fine, except that it required intensifying the lighting and created extra overhead from the walls.
